src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
```

Comment:
This is a related issue, but I think we ultimately need to go back to testing registered peers through the full domain-fronted stack instead of by direct IP. Peerscanner should be in charge of or at least should directly influence cloud.yaml. The way it stands now if a specific domain front stopped working, peerscanner would never know, and the cloud.yaml would not get updated. If we did the full domain fronted proxy call instead, peerscanner could automatically update the cloud.yaml to give clients a shot at continuing to function. 

On a related note, we should also start including the cloud.yaml URL in the cloud.yaml itself, as ideally we'd store it on an unblocked mirror and could change that mirror easily. That actually calls for going back to direct calls to the cloud.yaml and using it directly if it's not blocked to bypass the potential point of failure in domain-fronted calls.

Peerscanner will eventually also need to directly affect cloud.yaml by setting the percentage of traffic going to peers based on how many peers there are versus fallbacks at any given time.

I don't think there's anything we need to change immediately, but I'll ticket having peerscanner go back to how it used to do domain-fronted checks for fallbacks so it can update cloud.yaml directly.